client/request_test.go

```diff
@@ -28,24 +28,20 @@ func TestSetHostHeader(t *testing.T) {
 		expectedURLHost string
 	}{
 		{
-			"unix:///var/run/docker.sock",
-			"docker",
-			"/var/run/docker.sock",
+			host:            "unix:///var/run/docker.sock",
+			expectedURLHost: DummyHost,
 		},
 		{
-			"npipe:////./pipe/docker_engine",
-			"docker",
-			"//./pipe/docker_engine",
+			host:            "npipe:////./pipe/docker_engine",
+			expectedURLHost: DummyHost,
 		},
 		{
-			"tcp://0.0.0.0:4243",
-			"",
-			"0.0.0.0:4243",
+			host:            "tcp://0.0.0.0:4243",
+			expectedURLHost: "0.0.0.0:4243",
 		},
 		{
-			"tcp://localhost:4243",
-			"",
-			"localhost:4243",
+			host:            "tcp://localhost:4243",
+			expectedURLHost: "localhost:4243",
```

Comment:
`expectedHost` is always empty. 
Unused field.

Reply:
True, it's always empty in this variant; but it's also checked in the test; the test validates that it's empty.

I was also working on a different variant (after some discussion) where we use the `req.Host` field instead of manually overriding `req.URL.Host`. 

- https://github.com/moby/moby/pull/45964

That variant may actually be the more idiomatic approach as the `req.Host` field is _INTENDED_ for this purpose; https://github.com/golang/go/blob/go1.20.6/src/net/http/request.go#L223-L242

```go
	// For client requests, Host optionally overrides the Host
	// header to send. If empty, the Request.Write method uses
	// the value of URL.Host. Host may contain an international
	// domain name.
	Host string
```